Bottom: 756a9f6de0756b644f745f27429900bdf45db664
Top:    80dcd25e71af6ebd460b2d1055082a7434387a6c
Author: Allen Short <ashort@mozilla.com>
Date:   2018-07-09 13:20:36 -0500

Filter tables from schema browser (re #31)


---
diff --git a/client/app/components/queries/schema-browser.html b/client/app/components/queries/schema-browser.html
index 6e3f5180..fe7e2666 100644
--- a/client/app/components/queries/schema-browser.html
+++ b/client/app/components/queries/schema-browser.html
@@ -6,10 +6,20 @@
             ng-click="$ctrl.onRefresh()">
       <span class="zmdi zmdi-refresh"></span>
     </button>
+
+    <button class="btn btn-default"
+          title="Toggle Versioned Tables"
+          ng-click="$ctrl.flipToggleVersionedTables($ctrl.versionToggle, $ctrl.tabletogglestring)"
+          ng-if="$ctrl.tabletogglestring && $ctrl.tabletogglestring != ''"
+          >
+    <span class="fa " ng-class="{'fa-toggle-on': $ctrl.versionToggle == true, 'fa-toggle-off': !$ctrl.versionToggle}">
+      <input type="checkbox" id="versioned-tables-toggle" ng-model="$ctrl.versionToggle" hidden/>
+    </span>
+    </button>
   </div>
 
   <div class="schema-browser" vs-repeat vs-size="$ctrl.getSize(table)">
-    <div ng-repeat="table in $ctrl.schema | filter:$ctrl.schemaFilterObject track by table.name">
+    <div ng-repeat="table in $ctrl.schema | filter:$ctrl.schemaFilterObject | filter: {name: '!'+$ctrl.versionFilter}">
       <div class="table-name" ng-click="$ctrl.showTable(table)">
         <i class="fa fa-table"></i>
         <strong>
diff --git a/client/app/components/queries/schema-browser.js b/client/app/components/queries/schema-browser.js
index 34615aa5..ded89d09 100644
--- a/client/app/components/queries/schema-browser.js
+++ b/client/app/components/queries/schema-browser.js
@@ -3,6 +3,9 @@ import template from './schema-browser.html';
 function SchemaBrowserCtrl($rootScope, $scope) {
   'ngInject';
 
+  this.versionToggle = false;
+  this.versionFilter = 'abcdefghijklmnop';
+
   this.showTable = (table) => {
     table.collapsed = !table.collapsed;
     $scope.$broadcast('vsRepeatTrigger');
@@ -21,6 +24,15 @@ function SchemaBrowserCtrl($rootScope, $scope) {
   this.isEmpty = function isEmpty() {
     return this.schema === undefined || this.schema.length === 0;
   };
+  this.flipToggleVersionedTables = (versionToggle, toggleString) => {
+    if (versionToggle === false) {
+      this.versionToggle = true;
+      this.versionFilter = toggleString;
+    } else {
+      this.versionToggle = false;
+      this.versionFilter = 'abcdefghijklmnop';
+    }
+  };
 
   this.itemSelected = ($event, hierarchy) => {
     $rootScope.$broadcast('query-editor.command', 'paste', hierarchy.join('.'));
@@ -44,7 +56,9 @@ function SchemaBrowserCtrl($rootScope, $scope) {
 const SchemaBrowser = {
   bindings: {
     schema: '<',
+    tabletogglestring: '<',
     onRefresh: '&',
+    flipToggleVersionedTables: '&',
   },
   controller: SchemaBrowserCtrl,
   template,
diff --git a/client/app/pages/queries/query.html b/client/app/pages/queries/query.html
index 1744f9be..9641d4ac 100644
--- a/client/app/pages/queries/query.html
+++ b/client/app/pages/queries/query.html
@@ -96,7 +96,7 @@
       </div>
 
       <div class="editor__left__schema" ng-if="sourceMode">
-        <schema-browser class="schema-container" schema="schema" on-refresh="refreshSchema()"></schema-browser>
+        <schema-browser class="schema-container" schema="schema" tabletogglestring="dataSource.options.toggle_table_string" on-refresh="refreshSchema()"></schema-browser>
       </div>
       <div ng-if="!sourceMode" style="flex-grow: 1;">&nbsp;</div>
 
diff --git a/redash/query_runner/__init__.py b/redash/query_runner/__init__.py
index c252d841..2eb14128 100644
--- a/redash/query_runner/__init__.py
+++ b/redash/query_runner/__init__.py
@@ -171,6 +171,12 @@ class BaseHTTPQueryRunner(BaseQueryRunner):
                     'type': 'string',
                     'title': cls.password_title,
                 },
+                'toggle_table_string': {
+                    "type": "string",
+                    "title": "Toggle Table String",
+                    "default": "_v",
+                    "info": "This string will be used to toggle visibility of tables in the schema browser when editing a query in order to remove non-useful tables from sight."
+                },
             },
             'secret': ['password'],
             'order': ['url', 'username', 'password']
diff --git a/redash/query_runner/athena.py b/redash/query_runner/athena.py
index d9d27365..12633363 100644
--- a/redash/query_runner/athena.py
+++ b/redash/query_runner/athena.py
@@ -83,6 +83,12 @@ class Athena(BaseQueryRunner):
                     'title': 'Athena Work Group',
                     'default': 'primary'
                 },
+                "toggle_table_string": {
+                    "type": "string",
+                    "title": "Toggle Table String",
+                    "default": "_v",
+                    "info": "This string will be used to toggle visibility of tables in the schema browser when editing a query in order to remove non-useful tables from sight."
+                },
             },
             'required': ['region', 's3_staging_dir'],
             'order': ['region', 'aws_access_key', 'aws_secret_key', 's3_staging_dir', 'schema', 'work_group'],
diff --git a/redash/query_runner/axibase_tsd.py b/redash/query_runner/axibase_tsd.py
index 78f533fd..4514f886 100644
--- a/redash/query_runner/axibase_tsd.py
+++ b/redash/query_runner/axibase_tsd.py
@@ -132,6 +132,12 @@ class AxibaseTSD(BaseQueryRunner):
                 'trust_certificate': {
                     'type': 'boolean',
                     'title': 'Trust SSL Certificate'
+                },
+                "toggle_table_string": {
+                    "type": "string",
+                    "title": "Toggle Table String",
+                    "default": "_v",
+                    "info": "This string will be used to toggle visibility of tables in the schema browser when editing a query in order to remove non-useful tables from sight."
                 }
             },
             'required': ['username', 'password', 'hostname', 'protocol', 'port'],
diff --git a/redash/query_runner/big_query.py b/redash/query_runner/big_query.py
index 552b429c..22d77c88 100644
--- a/redash/query_runner/big_query.py
+++ b/redash/query_runner/big_query.py
@@ -126,6 +126,12 @@ class BigQuery(BaseQueryRunner):
                 'maximumBillingTier': {
                     "type": "number",
                     "title": "Maximum Billing Tier"
+                },
+                "toggle_table_string": {
+                    "type": "string",
+                    "title": "Toggle Table String",
+                    "default": "_v",
+                    "info": "This string will be used to toggle visibility of tables in the schema browser when editing a query in order to remove non-useful tables from sight."
                 }
             },
             'required': ['jsonKeyFile', 'projectId'],
diff --git a/redash/query_runner/cass.py b/redash/query_runner/cass.py
index 0f0c72ff..11550dd1 100644
--- a/redash/query_runner/cass.py
+++ b/redash/query_runner/cass.py
@@ -61,6 +61,12 @@ class Cassandra(BaseQueryRunner):
                     'type': 'number',
                     'title': 'Timeout',
                     'default': 10
+                },
+                "toggle_table_string": {
+                    "type": "string",
+                    "title": "Toggle Table String",
+                    "default": "_v",
+                    "info": "This string will be used to toggle visibility of tables in the schema browser when editing a query in order to remove non-useful tables from sight."
                 }
             },
             'required': ['keyspace', 'host']
diff --git a/redash/query_runner/clickhouse.py b/redash/query_runner/clickhouse.py
index 8a07b363..6603ce9e 100644
--- a/redash/query_runner/clickhouse.py
+++ b/redash/query_runner/clickhouse.py
@@ -36,6 +36,12 @@ class ClickHouse(BaseSQLQueryRunner):
                     "type": "number",
                     "title": "Request Timeout",
                     "default": 30
+                },
+                "toggle_table_string": {
+                    "type": "string",
+                    "title": "Toggle Table String",
+                    "default": "_v",
+                    "info": "This string will be used to toggle visibility of tables in the schema browser when editing a query in order to remove non-useful tables from sight."
                 }
             },
             "required": ["dbname"],
diff --git a/redash/query_runner/dynamodb_sql.py b/redash/query_runner/dynamodb_sql.py
index 5f4cc844..333989f2 100644
--- a/redash/query_runner/dynamodb_sql.py
+++ b/redash/query_runner/dynamodb_sql.py
@@ -47,6 +47,12 @@ class DynamoDBSQL(BaseSQLQueryRunner):
                 },
                 "secret_key": {
                     "type": "string",
+                },
+                "toggle_table_string": {
+                    "type": "string",
+                    "title": "Toggle Table String",
+                    "default": "_v",
+                    "info": "This string will be used to toggle visibility of tables in the schema browser when editing a query in order to remove non-useful tables from sight."
                 }
             },
             "required": ["access_key", "secret_key"],
diff --git a/redash/query_runner/elasticsearch.py b/redash/query_runner/elasticsearch.py
index 22e08d31..4d642ceb 100644
--- a/redash/query_runner/elasticsearch.py
+++ b/redash/query_runner/elasticsearch.py
@@ -62,6 +62,12 @@ class BaseElasticSearch(BaseQueryRunner):
                 'basic_auth_password': {
                     'type': 'string',
                     'title': 'Basic Auth Password'
+                },
+                "toggle_table_string": {
+                    "type": "string",
+                    "title": "Toggle Table String",
+                    "default": "_v",
+                    "info": "This string will be used to toggle visibility of tables in the schema browser when editing a query in order to remove non-useful tables from sight."
                 }
             },
             "order": ['server', 'basic_auth_user', 'basic_auth_password'],
diff --git a/redash/query_runner/google_analytics.py b/redash/query_runner/google_analytics.py
index 71be5220..117205a7 100644
--- a/redash/query_runner/google_analytics.py
+++ b/redash/query_runner/google_analytics.py
@@ -102,6 +102,12 @@ class GoogleAnalytics(BaseSQLQueryRunner):
                 'jsonKeyFile': {
                     "type": "string",
                     'title': 'JSON Key File'
+                },
+                "toggle_table_string": {
+                    "type": "string",
+                    "title": "Toggle Table String",
+                    "default": "_v",
+                    "info": "This string will be used to toggle visibility of tables in the schema browser when editing a query in order to remove non-useful tables from sight."
                 }
             },
             'required': ['jsonKeyFile'],
diff --git a/redash/query_runner/google_spreadsheets.py b/redash/query_runner/google_spreadsheets.py
index 3c22f45b..9c1f0fee 100644
--- a/redash/query_runner/google_spreadsheets.py
+++ b/redash/query_runner/google_spreadsheets.py
@@ -152,6 +152,12 @@ class GoogleSpreadsheet(BaseQueryRunner):
                 'jsonKeyFile': {
                     "type": "string",
                     'title': 'JSON Key File'
+                },
+                "toggle_table_string": {
+                    "type": "string",
+                    "title": "Toggle Table String",
+                    "default": "_v",
+                    "info": "This string will be used to toggle visibility of tables in the schema browser when editing a query in order to remove non-useful tables from sight."
                 }
             },
             'required': ['jsonKeyFile'],
diff --git a/redash/query_runner/graphite.py b/redash/query_runner/graphite.py
index 4e594c8b..adc08260 100644
--- a/redash/query_runner/graphite.py
+++ b/redash/query_runner/graphite.py
@@ -43,6 +43,12 @@ class Graphite(BaseQueryRunner):
                 'verify': {
                     'type': 'boolean',
                     'title': 'Verify SSL certificate'
+                },
+                "toggle_table_string": {
+                    "type": "string",
+                    "title": "Toggle Table String",
+                    "default": "_v",
+                    "info": "This string will be used to toggle visibility of tables in the schema browser when editing a query in order to remove non-useful tables from sight."
                 }
             },
             'required': ['url'],
diff --git a/redash/query_runner/hive_ds.py b/redash/query_runner/hive_ds.py
index b3c78bf4..6d8ec67c 100644
--- a/redash/query_runner/hive_ds.py
+++ b/redash/query_runner/hive_ds.py
@@ -55,6 +55,12 @@ class Hive(BaseSQLQueryRunner):
                 "username": {
                     "type": "string"
                 },
+                "toggle_table_string": {
+                    "type": "string",
+                    "title": "Toggle Table String",
+                    "default": "_v",
+                    "info": "This string will be used to toggle visibility of tables in the schema browser when editing a query in order to remove non-useful tables from sight."
+                }
             },
             "order": ["host", "port", "database", "username"],
             "required": ["host"]
@@ -98,14 +104,14 @@ class Hive(BaseSQLQueryRunner):
             database=self.configuration.get('database', 'default'),
             username=self.configuration.get('username', None),
         )
-        
+
         return connection
 
 
     def run_query(self, query, user):
         connection = None
         try:
-            connection = self._get_connection() 
+            connection = self._get_connection()
             cursor = connection.cursor()
 
             cursor.execute(query)
@@ -214,7 +220,7 @@ class HiveHttp(Hive):
 
         # create connection
         connection = hive.connect(thrift_transport=transport)
-        
+
         return connection
 
 
diff --git a/redash/query_runner/impala_ds.py b/redash/query_runner/impala_ds.py
index 5b8b5907..b57bb0b7 100644
--- a/redash/query_runner/impala_ds.py
+++ b/redash/query_runner/impala_ds.py
@@ -64,6 +64,12 @@ class Impala(BaseSQLQueryRunner):
                 },
                 "timeout": {
                     "type": "number"
+                },
+                "toggle_table_string": {
+                    "type": "string",
+                    "title": "Toggle Table String",
+                    "default": "_v",
+                    "info": "This string will be used to toggle visibility of tables in the schema browser when editing a query in order to remove non-useful tables from sight."
                 }
             },
             "required": ["host"],
diff --git a/redash/query_runner/influx_db.py b/redash/query_runner/influx_db.py
index 47f3a420..aee41318 100644
--- a/redash/query_runner/influx_db.py
+++ b/redash/query_runner/influx_db.py
@@ -57,6 +57,12 @@ class InfluxDB(BaseQueryRunner):
             'properties': {
                 'url': {
                     'type': 'string'
+                },
+                "toggle_table_string": {
+                    "type": "string",
+                    "title": "Toggle Table String",
+                    "default": "_v",
+                    "info": "This string will be used to toggle visibility of tables in the schema browser when editing a query in order to remove non-useful tables from sight."
                 }
             },
             'required': ['url']
diff --git a/redash/query_runner/memsql_ds.py b/redash/query_runner/memsql_ds.py
index bbec2836..b573b529 100644
--- a/redash/query_runner/memsql_ds.py
+++ b/redash/query_runner/memsql_ds.py
@@ -55,6 +55,12 @@ class MemSQL(BaseSQLQueryRunner):
                 },
                 "password": {
                     "type": "string"
+                },
+                "toggle_table_string": {
+                    "type": "string",
+                    "title": "Toggle Table String",
+                    "default": "_v",
+                    "info": "This string will be used to toggle visibility of tables in the schema browser when editing a query in order to remove non-useful tables from sight."
                 }
 
             },
diff --git a/redash/query_runner/mongodb.py b/redash/query_runner/mongodb.py
index 44511647..1dee9ac1 100644
--- a/redash/query_runner/mongodb.py
+++ b/redash/query_runner/mongodb.py
@@ -136,6 +136,12 @@ class MongoDB(BaseQueryRunner):
                     'type': 'string',
                     'title': 'Replica Set Name'
                 },
+                "toggle_table_string": {
+                    "type": "string",
+                    "title": "Toggle Table String",
+                    "default": "_v",
+                    "info": "This string will be used to toggle visibility of tables in the schema browser when editing a query in order to remove non-useful tables from sight."
+                },
             },
             'required': ['connectionString', 'dbName']
         }
diff --git a/redash/query_runner/mssql.py b/redash/query_runner/mssql.py
index 007aa825..0cf25567 100644
--- a/redash/query_runner/mssql.py
+++ b/redash/query_runner/mssql.py
@@ -60,6 +60,12 @@ class SqlServer(BaseSQLQueryRunner):
                 "db": {
                     "type": "string",
                     "title": "Database Name"
+                },
+                "toggle_table_string": {
+                    "type": "string",
+                    "title": "Toggle Table String",
+                    "default": "_v",
+                    "info": "This string will be used to toggle visibility of tables in the schema browser when editing a query in order to remove non-useful tables from sight."
                 }
             },
             "required": ["db"],
diff --git a/redash/query_runner/mysql.py b/redash/query_runner/mysql.py
index 9c674aea..9815100b 100644
--- a/redash/query_runner/mysql.py
+++ b/redash/query_runner/mysql.py
@@ -54,6 +54,12 @@ class Mysql(BaseSQLQueryRunner):
                 'port': {
                     'type': 'number',
                     'default': 3306,
+                },
+                "toggle_table_string": {
+                    "type": "string",
+                    "title": "Toggle Table String",
+                    "default": "_v",
+                    "info": "This string will be used to toggle visibility of tables in the schema browser when editing a query in order to remove non-useful tables from sight."
                 }
             },
             "order": ['host', 'port', 'user', 'passwd', 'db'],
diff --git a/redash/query_runner/oracle.py b/redash/query_runner/oracle.py
index eff92500..8979ebb1 100644
--- a/redash/query_runner/oracle.py
+++ b/redash/query_runner/oracle.py
@@ -63,6 +63,12 @@ class Oracle(BaseSQLQueryRunner):
                 "servicename": {
                     "type": "string",
                     "title": "DSN Service Name"
+                },
+                "toggle_table_string": {
+                    "type": "string",
+                    "title": "Toggle Table String",
+                    "default": "_v",
+                    "info": "This string will be used to toggle visibility of tables in the schema browser when editing a query in order to remove non-useful tables from sight."
                 }
             },
             "required": ["servicename", "user", "password", "host", "port"],
diff --git a/redash/query_runner/pg.py b/redash/query_runner/pg.py
index 5524375d..3c842400 100644
--- a/redash/query_runner/pg.py
+++ b/redash/query_runner/pg.py
@@ -95,6 +95,12 @@ class PostgreSQL(BaseSQLQueryRunner):
                    "type": "string",
                    "title": "SSL Mode",
                    "default": "prefer"
+                },
+                "toggle_table_string": {
+                    "type": "string",
+                    "title": "Toggle Table String",
+                    "default": "_v",
+                    "info": "This string will be used to toggle visibility of tables in the schema browser when editing a query in order to remove non-useful tables from sight."
                 }
             },
             "order": ['host', 'port', 'user', 'password'],
diff --git a/redash/query_runner/presto.py b/redash/query_runner/presto.py
index d4c919d3..17a797ca 100644
--- a/redash/query_runner/presto.py
+++ b/redash/query_runner/presto.py
@@ -59,6 +59,12 @@ class Presto(BaseQueryRunner):
                 'password': {
                     'type': 'string'
                 },
+                "toggle_table_string": {
+                    "type": "string",
+                    "title": "Toggle Table String",
+                    "default": "_v",
+                    "info": "This string will be used to toggle visibility of tables in the schema browser when editing a query in order to remove non-useful tables from sight."
+                },
             },
             'order': ['host', 'protocol', 'port', 'username', 'password', 'schema', 'catalog'],
             'required': ['host']
diff --git a/redash/query_runner/python.py b/redash/query_runner/python.py
index f6cc2fbc..4fdf0de6 100644
--- a/redash/query_runner/python.py
+++ b/redash/query_runner/python.py
@@ -55,6 +55,12 @@ class Python(BaseQueryRunner):
                 },
                 'additionalModulesPaths': {
                     'type': 'string'
+                },
+                "toggle_table_string": {
+                    "type": "string",
+                    "title": "Toggle Table String",
+                    "default": "_v",
+                    "info": "This string will be used to toggle visibility of tables in the schema browser when editing a query in order to remove non-useful tables from sight."
                 }
             },
         }
diff --git a/redash/query_runner/salesforce.py b/redash/query_runner/salesforce.py
index 527f1e26..7222028f 100644
--- a/redash/query_runner/salesforce.py
+++ b/redash/query_runner/salesforce.py
@@ -81,6 +81,12 @@ class Salesforce(BaseQueryRunner):
                     "type": "string",
                     "title": "Salesforce API Version",
                     "default": DEFAULT_API_VERSION
+                },
+                "toggle_table_string": {
+                    "type": "string",
+                    "title": "Toggle Table String",
+                    "default": "_v",
+                    "info": "This string will be used to toggle visibility of tables in the schema browser when editing a query in order to remove non-useful tables from sight."
                 }
             },
             "required": ["username", "password", "token"],
diff --git a/redash/query_runner/script.py b/redash/query_runner/script.py
index 38e3ae62..1a4b80bd 100644
--- a/redash/query_runner/script.py
+++ b/redash/query_runner/script.py
@@ -49,6 +49,12 @@ class Script(BaseQueryRunner):
                 'shell': {
                     'type': 'boolean',
                     'title': 'Execute command through the shell'
+                },
+                "toggle_table_string": {
+                    "type": "string",
+                    "title": "Toggle Table String",
+                    "default": "_v",
+                    "info": "This string will be used to toggle visibility of tables in the schema browser when editing a query in order to remove non-useful tables from sight."
                 }
             },
             'required': ['path']
diff --git a/redash/query_runner/snowflake.py b/redash/query_runner/snowflake.py
index f6ff5100..b2c2c21b 100644
--- a/redash/query_runner/snowflake.py
+++ b/redash/query_runner/snowflake.py
@@ -49,7 +49,13 @@ class Snowflake(BaseQueryRunner):
                 "region": {
                     "type": "string",
                     "default": "us-west"
-                }
+                },
+                "toggle_table_string": {
+                    "type": "string",
+                    "title": "Toggle Table String",
+                    "default": "_v",
+                    "info": "This string will be used to toggle visibility of tables in the schema browser when editing a query in order to remove non-useful tables from sight."
+                },
             },
             "order": ["account", "user", "password", "warehouse", "database", "region"],
             "required": ["user", "password", "account", "database", "warehouse"],
diff --git a/redash/query_runner/sqlite.py b/redash/query_runner/sqlite.py
index c1933d81..9f02315e 100644
--- a/redash/query_runner/sqlite.py
+++ b/redash/query_runner/sqlite.py
@@ -21,6 +21,12 @@ class Sqlite(BaseSQLQueryRunner):
                 "dbpath": {
                     "type": "string",
                     "title": "Database Path"
+                },
+                "toggle_table_string": {
+                    "type": "string",
+                    "title": "Toggle Table String",
+                    "default": "_v",
+                    "info": "This string will be used to toggle visibility of tables in the schema browser when editing a query in order to remove non-useful tables from sight."
                 }
             },
             "required": ["dbpath"],
diff --git a/redash/query_runner/treasuredata.py b/redash/query_runner/treasuredata.py
index 5e3673ed..52ee2029 100644
--- a/redash/query_runner/treasuredata.py
+++ b/redash/query_runner/treasuredata.py
@@ -58,6 +58,12 @@ class TreasureData(BaseQueryRunner):
                     'type': 'boolean',
                     'title': 'Auto Schema Retrieval',
                     'default': False
+                },
+                "toggle_table_string": {
+                    "type": "string",
+                    "title": "Toggle Table String",
+                    "default": "_v",
+                    "info": "This string will be used to toggle visibility of tables in the schema browser when editing a query in order to remove non-useful tables from sight."
                 }
             },
             'required': ['apikey','db']
diff --git a/redash/query_runner/vertica.py b/redash/query_runner/vertica.py
index 92ab864c..92633eb2 100644
--- a/redash/query_runner/vertica.py
+++ b/redash/query_runner/vertica.py
@@ -60,6 +60,12 @@ class Vertica(BaseSQLQueryRunner):
                     "type": "number",
                     "title": "Connection Timeout"
                 },
+                "toggle_table_string": {
+                    "type": "string",
+                    "title": "Toggle Table String",
+                    "default": "_v",
+                    "info": "This string will be used to toggle visibility of tables in the schema browser when editing a query in order to remove non-useful tables from sight."
+                },
             },
             'required': ['database'],
             'order': ['host', 'port', 'user', 'password', 'database', 'read_timeout', 'connection_timeout'],
@@ -117,7 +123,7 @@ class Vertica(BaseSQLQueryRunner):
                 'database': self.configuration.get('database', ''),
                 'read_timeout': self.configuration.get('read_timeout', 600)
             }
-            
+
             if self.configuration.get('connection_timeout'):
                 conn_info['connection_timeout'] = self.configuration.get('connection_timeout')
 
diff --git a/tests/test_cli.py b/tests/test_cli.py
index 25ab3ef5..0546272c 100644
--- a/tests/test_cli.py
+++ b/tests/test_cli.py
@@ -16,7 +16,7 @@ class DataSourceCommandTests(BaseTestCase):
         result = runner.invoke(
             manager,
             ['ds', 'new'],
-            input="test\n%s\n\n\nexample.com\n\n\ntestdb\n" % (pg_i,))
+            input="test\n%s\n\n\n\n\nexample.com\n\n\ntestdb\n" % (pg_i,))
         self.assertFalse(result.exception)
         self.assertEqual(result.exit_code, 0)
         self.assertEqual(DataSource.query.count(), 1)
